Backend/scala-backend/app/controllers/SurveyController.scala

```diff
@@ -146,18 +172,119 @@ class SurveyController @Inject() (
             .getStatus(classId)
             .flatMap(status => {
               if (status == SurveyStatus.Open) {
-                classModel
-                  .updateStatus(classId, SurveyStatus.Closed)
-                  .map(_ =>
-                    Ok(Json.obj("message" -> "success - survey closed"))
-                  )
-              } else Future.successful(Gone("Survey has wrong status"))
+                request.body.asJson match {
+                  case Some(content) => {
+                    val mergingJsonOption: JsResult[MergeInterface] =
+                      Json.fromJson[MergeInterface](content)
+
+                    if (mergingJsonOption.isSuccess) {
+                      val mergingSuccess: Future[Boolean] =
+                        mergeStudents(classId, mergingJsonOption.get)
+
+                      mergingSuccess.flatMap(success => {
+                        classModel
+                          .updateStatus(classId, SurveyStatus.Closed)
+                          .map(_ =>
+                            Ok(
+                              Json.obj("message" -> "success - survey closed")
+                            ) //return
+                          )
+                      })
+                    } else {
+                      Future.successful(
+                        UnsupportedMediaType("Wrong JSON format") //return
+                      )
+                    }
+                  }
+                  case None =>
+                    Future.successful(BadRequest("Empty Body")) //return
+                }
+
+              } else Future.successful(Gone("Survey has wrong status")) //return
             })
         }
       }
       auth.withTeacherAuthentication(body)
   }
 
+  def mergeStudents(
+      classId: Int,
+      mergingObject: MergeInterface
+  ): Future[Boolean] = {
+
+    //1. create new
+    val creationSuccessList: Seq[Future[Boolean]] =
+      mergingObject.studentsToAdd.map(student =>
+        createStudents(classId, student, Seq())
+      )
+
+    val creationSuccess: Future[Boolean] = Future
+      .sequence(creationSuccessList)
+      .map((arr: Seq[Boolean]) => {
+        arr.forall(_ == true)
+      })
+
+    // 2. rename
+    creationSuccess.flatMap(cSuccess => {
```

Comment:
flatten out
on the LHS of 228: val renameSuccess: Future[Boolean] = creationSucess.flatMap(...
...
return on line 238
})